lib/hmmlearn/vhmm.py

```diff
@@ -0,0 +1,611 @@
+
+    @property
+    def means_(self):
+        """
+        Compat for _BaseGaussianHMM
+        """
+        return self.means_posterior_
+
+    def _init(self, X, lengths):
+        """
+        Initialize model parameters prior to fitting.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Feature matrix of individual samples.
+        lengths : array-like of integers, shape (n_sequences, )
+            Lengths of the individual sequences in ``X``. The sum of
+            these should be ``n_samples``.
+        """
+        _check_and_set_n_features(self, X)
+        super()._init(X, lengths)
+
+        X_mean = X.mean(axis=0)
+        # Kmeans will be used for initializing both the means
+        # and the covariances
+        kmeans = cluster.KMeans(n_clusters=self.n_components,
+                                random_state=self.random_state)
+        kmeans.fit(X)
+        cluster_counts = np.bincount(kmeans.predict(X))
+
+        if self._needs_init("m", "means_posterior_"):
+            if self.means_prior is None:
+                self.means_prior_ = np.full(
+                    (self.n_components, self.n_features), X_mean)
+            else:
+                self.means_prior_ = self.means_prior
+            # Initialize to the data means
+            self.means_posterior_ = np.copy(kmeans.cluster_centers_)
+
+        if self._needs_init("b", "beta_posterior_"):
+            if self.beta_prior is None:
+                self.beta_prior_ = np.zeros(self.n_components) + 1
+            else:
+                self.beta_prior_ = self.beta_prior
+
+            # Count of items in each cluster
+            self.beta_posterior_ = np.copy(cluster_counts)
+
+        if self._needs_init("d", "dof_posterior_"):
+            if self.covariance_type == "full":
+                if self.dof_prior is None:
+                    self.dof_prior_ = np.full(
+                        (self.n_components,), self.n_features)
+                else:
+                    self.dof_prior_ = self.dof_prior
+                self.dof_posterior_ = np.copy(cluster_counts)
+
+            elif self.covariance_type == "tied":
+                if self.dof_prior is None:
+                    self.dof_prior_ = self.n_features
+                else:
+                    self.dof_prior_ = self.dof_prior
+
+                self.dof_posterior_ = cluster_counts.sum()
+            elif self.covariance_type == "diag":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            elif self.covariance_type == "spherical":
+                raise NotImplementedError(
+                    f"{self.covariance_type} is not implemented yet"
+                )
+            else:
+                raise ValueError(
+                    f"Unknown covariance_type: {self.covariance_type}")
+
+        if self._needs_init("w", "W_k_"):
```

Comment:
Are these W_foo attributes that the end user is supposed to sometimes want to set themselves?  If so they should be documented (at least cursorily), otherwise they should be private.

Reply:
Work in progress, and will actually remove a set of `W` for simplicity.

Reply:
I will work on this next.

Reply:
I've renamed these to match what I found in a reference `W-> scale`